platforms/enterprise/enterprise-operations/src/main/java/org/gradle/operations/problems/ProblemUsageProgressDetails.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.operations.problems;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * A problem from the problem API reported via a progress event.
+ *
+ * @since 8.14
+ */
+public interface ProblemUsageProgressDetails {
```

Comment:
💭 I wonder if we should capture as well whether the stacktrace is important for the problem or not. The user defines this in the problem spec.

Reply:
if the stacktrace is not important, should we just transmit a null failure there ?
Or, otherwise, what is the point of having the failure and the fact that it's not important, what will we do with it ?

Reply:
We could do that. So then we would lose the option to actually show the stacktrace in the UI on demand for diagnostics.

Reply:
> show the stacktrace in the UI on demand for diagnostics
you mean for us, with a secret flag or sth or you mean that 
stacktrace is important => failure is always shown
stacktrace is important => failure is not shown (eg collapsed) but accessible for users via a button, or sth